### PR TITLE
Add semantic segmentation documentation

### PR DESCRIPTION
**Pull request by guarin**
## What has changed and why?

* Add documentation for semantic segmentation

## How has it been tested?

[Semantic Segmentation - LightlyTrain documentation - 3.pdf](https://github.com/user-attachments/files/21578518/Semantic.Segmentation.-.LightlyTrain.documentation.-.3.pdf)

[LightlyTrain documentation - semantic segmentation.pdf](https://github.com/user-attachments/files/21544247/LightlyTrain.documentation.-.semantic.segmentation.pdf)

## Did you update [CHANGELOG.md](../CHANGELOG.md)?

- [x] Yes
- [ ] Not needed (internal change)

## Did you update the documentation?

- [x] Yes
- [ ] Not needed (internal change without effects for user)

### CODE REVIEW

File: docs/source/semantic_segmentation.md [hunk 1]
```diff
@@ -0,0 +1,317 @@
+(semantic-segmentation)=
+
+# Semantic Segmentation
```
Comment (IgorSusmelj) [inline docs/source/semantic_segmentation.md:3]: There is information missing about the input format. How do I need to provide the masks? How are they encoded? png images? If yes, what RGB values do they need to have? Do the masks need to have the same folder structure as the images? Same filenames? We should assume here that the reason has no clue and just quickly wants to train a model. Can we point the reader to a reference dataset or so with example images with correct masks?

Other general questions we should clarify:
- how can I change the input/output resolution of the model? Does it support non square images?

Comment (IgorSusmelj) [reply docs/source/semantic_segmentation.md:3]: The tutorial is quite long. Some people might only read the first part.

An easy to clarify these open questions would be to add two bullet points:
- the input format follows ADE20k which uses RGB masks saved as .png files [link here to an example dataset]
- the model supports arbitrary input/output resolutions

Comment (liopeer) [reply docs/source/semantic_segmentation.md:3]: > There is information missing about the input format. How do I need to provide the masks? How are they encoded? png images? If yes, what RGB values do they need to have? Do the masks need to have the same folder structure as the images? Same filenames? We should assume here that the reason has no clue and just quickly wants to train a model.

This is in the `Data` section, which is imo good and in harmony with the documentation of the `lightly_train.train` command. But the `Data` section should come directly after the the initial code samples. I would therefore move the `Optional: Pretrain and Fine-tune a Semantic Segmentation Model` to the end of the page and just reference it at the current location through a `seealso`. That would also make it shorter and partially address:
> The tutorial is quite long. Some people might only read the first part.

> Can we point the reader to a reference dataset or so with example images with correct masks?

Fully agree that we need this. To keep consistency with the rest of the documentation however, I would prefer seeing that in the `Quickstart`. Will this make the Quickstart too bloated?

File: docs/source/semantic_segmentation.md [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+`ignore_classes` argument. The trained model will then not predict these classes.
+
+(semantic-segmentation-model)=
+
+## Model
+
+The `model` argument defines the model used for semantic segmentation training. The
+following models are available:
+
+- `dinov2/vits14-eomt`
```
Comment (liopeer) [inline docs/source/semantic_segmentation.md:188]: We should mention our own checkpoints (that we will upload) here as well. Can you also load the official EoMT weights? If yes, we should maybe add these as well?

File: docs/source/semantic_segmentation.md [hunk 3]
```diff
@@ -0,0 +1,317 @@
...
+
+````{tab} Python
+```python
+logger_args={"tensorboard": None}
+```
+````
+
+### Performance Optimizations
+
+For performance optimizations, e.g. using accelerators, multi-GPU, multi-node, and
```
Comment (liopeer) [inline docs/source/semantic_segmentation.md]: Do we log things like `data_time`, `batch_time` etc. as mentioned in the *Performance* docs? Is this automatically logged in Lightning Fabric as well? If not, I would probably just leave the performance section away for now.

File: docs/source/semantic_segmentation.md (hunk 2 above)
Comment (guarin) [reply docs/source/semantic_segmentation.md:188]: Will do that once they are ready

File: docs/source/semantic_segmentation.md (hunk 3 above)
Comment (guarin) [reply docs/source/semantic_segmentation.md]: Removed
